### PR TITLE
#2 removed neuro oreintation and frites dependency

### PR DESCRIPTION
**Pull request by Dishie2498**
Added modules core and metrics - generalised functions for perspectives outside of neuroscience

### CODE REVIEW

**Comment by EtienneCmb** [comment]
Thanks @Dishie2498, great work !

Could please add the following :

# core/it.py

Copnorm functions are related to information theoretical measures. Let's cut `ctransform`, `copnorm_1d` and `copnorm_nd` from the oinfo file to `it.py`

# metrics/oinfo_zerolag.py

- Rename `conn_oinfo_jax` function to `oinfo_zerolag`

# core/__init__.py

Import functions :

```python
from .combinatory  import combinations  # noqa
from .it import ent_g,  copnorm_1d, copnorm_nd  # noqa
```

# metrics/__init__.py

```python
from .oinfo_zerolag import oinfo_zerolag  # noqa
```

File: hoi/metrics/oinfo_zerolag.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+def copnorm_nd(x, axis=-1):  # frites.core
+    assert isinstance(x, np.ndarray) and (x.ndim >= 1)
+    return np.apply_along_axis(copnorm_1d, axis, x)
+
+
+def conn_oinfo_jax(
+    data,
+    y=None,
+    variables=None,
```
Comment (EtienneCmb) [inline hoi/metrics/oinfo_zerolag.py:17]: Let's remove the inputs `variables` and `features` to simplify even more

File: hoi/metrics/oinfo_zerolag.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    features=None,
+    minsize=3,
+    maxsize=5,
+    sort=True,
```
Comment (EtienneCmb) [inline hoi/metrics/oinfo_zerolag.py:21]: Is the sorting used somewhere?

File: hoi/metrics/oinfo_zerolag.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        synergistic dominated interactions.
+    """
+    # ________________________________ INPUTS _________________________________
+    # inputs conversion
+    is_task_related = isinstance(y, (str, list, np.ndarray, tuple))
+    # kw_links = {"directed": False, "net": False}
+
+    # extract variables
+    # x, attrs = data.data, cfg["attrs"]
+    y, features, variables = (
```
Comment (EtienneCmb) [inline hoi/metrics/oinfo_zerolag.py:53]: This block is not essential since the input type is not an `xarray.DataArray` anymore. We can remove it.

**Review by EtienneCmb — CHANGES_REQUESTED** [review]
We can simplify the `oinfo_zerolag` even more

File: hoi/metrics/oinfo_zerolag.py (hunk 2 above)
Comment (Dishie2498) [reply hoi/metrics/oinfo_zerolag.py:21]: Yes. The function `combinations` is called within `oinfo_zerolag` which has a parameter `sort` to create a lexicographic order of the regions.

File: hoi/metrics/oinfo_zerolag.py (hunk 3 above)
Comment (Dishie2498) [reply hoi/metrics/oinfo_zerolag.py:53]:
```
y = data[0]
features = data[1]
variables = data[2]
```
So, are you suggesting to do this instead?

File: hoi/metrics/oinfo_zerolag.py (hunk 2 above)
Comment (EtienneCmb) [reply hoi/metrics/oinfo_zerolag.py:21]: Indeed. We could remove the `feature` input of the `combination` function, such as the sorting. We can do the sorting later, once we identify significant multiplets we could build the sorted name of the multiplets.

File: hoi/metrics/oinfo_zerolag.py (hunk 3 above)
Comment (EtienneCmb) [reply hoi/metrics/oinfo_zerolag.py:53]: The input data is going to be a standard NumPy array. You will not be able to access y, features etc. from the coordinates of a standard NumPy array.

File: hoi/metrics/oinfo_zerolag.py [hunk 4]
```diff
@@ -11,16 +12,22 @@
...
-    data,
-    y=None,
-    variables=None,
-    features=None,
-    minsize=3,
-    maxsize=5,
-    sort=True,
-    verbose=None,
-):
+def combin(n, k, task_related=False, sort=True):
```
Comment (EtienneCmb) [inline hoi/metrics/oinfo_zerolag.py:15]: You can directly use the one in `hoi/core/combinatory/combinations`

File: hoi/metrics/oinfo_zerolag.py [hunk 5]
```diff
@@ -46,16 +53,10 @@ def oinfo_zerolag(
...
 
     # extract variables
-    # x, attrs = data.data, cfg["attrs"]
-    y, features, variables = (
-        data["y"].data,
-        data["features"].data,
-        data["variables"].data,
-    )
-    n_features = len(features)
+    x = data
```
Comment (EtienneCmb) [inline hoi/metrics/oinfo_zerolag.py:58]: Change input name for `x` or internal code for `data`

**Comment by EtienneCmb** [comment]
Thank you @Dishie2498 , great work, the code is much easier to read now !

**Pull request merged**